File: number_of_steps_to_reduce_to_zero/typescript/nostrtz.ts
export function nostrtz(num: number): number {
  let steps: number = 0;
  while (num > 0) {
    let bw: number = num & 1
    if (bw === 0) {
      num = num >> 1
    }
    else {
      num = num - 1
    }
    steps++
  }
  return(steps)
}


export function nostrtz2(num: number): number {
  let steps: number = 0;
  while (num > 0) {
    if (num % 2 === 0) {
      num = num / 2
    }
    else {
      num = num - 1
    }
    steps++
  }
  return(steps)
}

File: 1342_Number_of_steps_to_reduce_to_zero/python/nostrtz.py
class Solution1:
    def num_of_steps(self, num: int):
        steps = 0
        while num > 0:
            if num % 2 == 0:
                num = num / 2
            else:
                num = num - 1

            steps += 1

        return (steps)


class Solution2:
    def num_of_steps(self, num: int):
        steps = 0
        while num > 0:
            if num & 1 == 0:
                num = num >> 1
            else:
                num = num - 1

            steps += 1

        return (steps)
